image/mirror.go

```diff
@@ -25,6 +28,14 @@ func Mirror(image string) string {
 	image = strings.Replace(image, "coredns/", "rancher/coredns-", 1)
 	image = strings.Replace(image, "minio/", "rancher/minio-", 1)
 
+	if !strings.HasPrefix(image, "rancher/") {
+		panic("image not prefixed with rancher/")
+	}
+
+	imageNameTag := strings.Split(image, ":")
+	if strings.HasSuffix(imageNameTag[0], "-") {
+		panic(fmt.Sprintf("Image [%s] has trailing '-', probably an error in image subsitution", image))
+	}
```

Comment:
I dont think we should be adding checks here and panic on failure - There is another API for kontainer-driver-metadata @kinarashah has added that calls this - that API handler needs to be able to handle the error and add suitable response.

Reply:
This makes sure that the resulting images are valid, there is no case to proceed if the resulting images are invalid. I'm fine with moving the check, perhaps in https://github.com/rancher/rancher/blob/master/pkg/image/export/main.go?

Reply:
I think the place to add this will be the download link on kontainerdrivermetadata -https://github.com/rancher/rancher/blob/master/pkg/api/customization/kontainerdriver/actionhandler.go#L94

Reply:
I put it in here to make this one fail so that the build fails when the images are not correct, versus discovering it when its too late, see https://github.com/rancher/rancher/blob/master/scripts/package#L42

Will putting it there prevent it from being built/released?

Reply:
@superseb true, but since now the method is used twice - during build + the download images link after running server, I'd think we can get check and panic here: 
https://github.com/rancher/rancher/blob/master/pkg/image/export/main.go#L68 
but leaving the mirror function as it is

Reply:
Will change, thanks @prachidamle @kinarashah